src/Core/Content/Product/Aggregate/ProductPrice/ProductPriceCollection.php

```diff
@@ -34,6 +34,13 @@ public function sortByQuantity(): void
         });
     }
 
+    public function sortByPrice(): void
+    {
+        $this->sort(function (ProductPriceEntity $a, ProductPriceEntity $b) {
+            return $a->getPrice()->first()->getGross() <=> $b->getPrice()->first()->getGross();
```

Comment:
Gross prices may not always be the decisive criterion. You can call `Context::getTaxState()` to find whether gross or net prices should be used. Depending on this then either sort by net or gross prices. To do so please change the signature of the method to `public function sortByPrice(Context $context): void` and also consider the tax state in the aforementioned unit test.

Reply:
@dneustadt Thanks for the hint. What would be the flow then if the tax is TAX_STATE_FREE ? Should it default to gross in that case?